internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go

```diff
@@ -19,6 +19,8 @@ import (
 	"github.com/hashicorp/terraform-provider-azurerm/utils"
 )
 
+var hostpoolResourceType = "azurerm_virtual_desktop_host_pool"
```

Comment:
we also need to lock on this in the Create, Update and Delete methods of this resource?

Reply:
also I think `registration_info` needs to become computed in this resource, else there'll be a perpectual conflict?

Reply:
> also I think `registration_info` needs to become computed in this resource, else there'll be a perpectual conflict?

The intent here was to avoid a breaking change - the docs for both the hostpool and the registration info make it clear  one or the other should be used - but not both, otherwise expect oddities!

WDYT? Is that enough or do we need to change to computed?

Reply:
Added locks to `virtual_desktop_host_pool` resource

Reply:
This will need to be computed once https://github.com/hashicorp/terraform-provider-azurerm/pull/14953 is merged - the latest API version changed the API behaviour from returning it in the original payload to using the secondary API